File: BE/Models/ViewModels/CarTypeDetailVM.cs
namespace GoWheels_WebAPI.Models.ViewModels
{
    public class CarTypeDetailVM
    {
        public int Id { get; set; }
        public int CarTypeId { get; set; }
        public string? CarTypeName { get; set; }
        public int CompanyId { get; set; }
        public string? CompanyName { get; set; }
        public bool IsDeleted { get; set; }
    }
}

File: FE/src/Model/CarTypeDetailVM.js
// src/models/CarTypeDetailVM.js

export default class CarTypeDetailVM {
    constructor(data = {}) {
      this.id = data.id || 0;
      this.carTypeId = data.carTypeId || 0;
      this.carTypeName = data.carTypeName || '';
      this.companyId = data.companyId || 0;
      this.companyName = data.companyName || '';
      this.isDeleted = data.isDeleted || false;
    }
  }
  